core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -569,10 +569,11 @@ sealed abstract class Chunk[+A] extends ChunkLike[A] with Serializable { self =>
     exists
   }
 
-  override final def hashCode: Int = toArrayOption match {
-    case None        => Seq.empty[A].hashCode
-    case Some(array) => array.toSeq.hashCode
-  }
+  override final def hashCode: Int =
+    toArrayOrNull match {
+      case null  => Seq.empty[A].hashCode
+      case array => array.toSeq.hashCode
```

Comment:
We can avoid the array conversion (which allocates another array):

```scala
import scala.util.hashing.MurmurHash3

MurmurHash3.orderedHash(array, MurmurHash3.seqSeed)
```

Reply:
Done